src/realm/transaction.cpp

```diff
@@ -933,9 +933,9 @@ void Transaction::set_transact_stage(DB::TransactStage stage) noexcept
 
 class NodeTree {
 public:
-    NodeTree(size_t evac_limit, size_t& work_limit)
+    NodeTree(size_t evac_limit, size_t work_limit)
         : m_evac_limit(evac_limit)
-        , m_work_limit(work_limit)
+        , m_work_limit(int64_t(work_limit))
```

Comment:
why do we want to allow for a negative work limit?

Reply:
Because then we can just subtract the sizes and just theck if the work_limit is not negative